src/auth/authenticator/oicd/authenticator.go

```diff
@@ -0,0 +1,70 @@
+package oicd
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/consensys/quorum-key-manager/pkg/tls/certificate"
+	"github.com/consensys/quorum-key-manager/src/auth/authenticator"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+const AuthMode = "JWT"
+
+type Authenticator struct {
+	jwtCheckers []*JWTChecker
+}
+
+var _ authenticator.Authenticator = Authenticator{}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{
+		jwtCheckers: []*JWTChecker{},
+	}
+
+	for _, cert := range cfg.Certificates {
+		keyPair, err := certificate.X509KeyPair([]byte(cert), nil)
+		if err != nil {
+			return nil, err
```

Comment:
I'm not sure but I feel that the error handling here should return an error from the `errors` package and log?

Reply:
The authenticator component will convert whatever checker error into a UnauhorizedError
```
	claims, err := a.jwtChecker.Check(req.Context(), token)
	if err != nil {
		return nil, errors.UnauthorizedError(err.Error())
	}
	```